src/main/java/com/devonfw/quarkus/productmanagement/service/v1/ProductRestService.java

```diff
@@ -20,92 +20,54 @@
-  @APIResponse(responseCode = "500") })
-  @Operation(operationId = "Get Products", description = "Returns list of Products matching given criteria, uses pagination")
-  @GET
-  // REST service methods should not declare exceptions, any thrown error will be transformed by exceptionMapper in
-  // tkit-rest
-  // We did not define custom @Path - so it will use class level path
-  public Page<ProductDto> getAll(@BeanParam ProductSearchCriteriaDto dto) {
-
-    return this.ucFindProduct.findProducts(dto);
-  }
-
-  @GET
-  @Path("criteriaApi")
-  public Page<ProductDto> getAllCriteriaApi(@BeanParam ProductSearchCriteriaDto dto) {
-
-    return this.ucFindProduct.findProductsByCriteriaApi(dto);
-  }
-
-  @GET
-  @Path("queryDsl")
-  public Page<ProductDto> getAllQueryDsl(@BeanParam ProductSearchCriteriaDto dto) {
-
-    return this.ucFindProduct.findProductsByQueryDsl(dto);
-  }
+  UcDelete ucDelete;
 
-  @GET
-  @Path("query")
-  public Page<ProductDto> getAllQuery(@BeanParam ProductSearchCriteriaDto dto) {
-
-    return this.ucFindProduct.findProductsByTitleQuery(dto);
-  }
+  @Inject
+  UcSave ucSave;
 
-  @GET
-  @Path("nativeQuery")
-  public Page<ProductDto> getAllNativeQuery(@BeanParam ProductSearchCriteriaDto dto) {
+  @POST
+  @Path("search")
+  public Page<ProductDto> getAllQueryDsl(@BeanParam ProductSearchCriteriaDto productSearch) {
 
-    return this.ucFindProduct.findProductsByTitleNativeQuery(dto);
+    return this.ucFindProduct.findProducts(productSearch);
   }
 
   @GET
-  @Path("ordered")
   public Page<ProductDto> getAllOrderedByTitle() {
 
-    return this.ucFindProduct.findProductsOrderedByTitle();
+    return this.ucFindProduct.findProductsOrderByTitle();
   }
 
   @APIResponses({
-  @APIResponse(responseCode = "200", description = "OK, New Product created", content = @Content(schema = @Schema(implementation = NewProductDto.class))),
+  @APIResponse(responseCode = "201", description = "OK, New Product created", content = @Content(schema = @Schema(implementation = ProductDto.class))),
   @APIResponse(responseCode = "400", description = "Client side error, invalid request"),
   @APIResponse(responseCode = "500") })
-  @Operation(operationId = "createNewProduct", description = "Stores new Product in DB")
+  @Operation(operationId = "save", description = "Stores new Product in DB")
   @POST
-  // We did not define custom @Path - so it will use class level path.
-  // Although we now have 2 methods with same path, it is ok, because it is a different method (get vs post)
-  public ProductDto createNewProduct(NewProductDto dto) {
+  public Response save(ProductDto product) {
 
-    return this.ucManageProduct.saveProduct(dto);
+    if (StringUtils.isEmpty(product.getTitle())) {
+      throw new WebApplicationException("Title was not set on request.", 400);
+    }
+    this.ucSave.saveProduct(product);
+    return Response.ok().status(201).build();
```

Comment:
201 created should always return an href to the newly created resource as `Location` response header (see `Jaxrs UriBuilder`) if it has no body. Otherwise the client asked to create a new product, but received no information that would allow it to query it afterwards

Reply:
I see some issue here:
1. The returned URL is incorrect as it is leading to 404 and omits the service path.
2. In devonfw we strictly voted against pointless boilerplate code due to RESTful and HATOES stuff. I have never ever seen a reasonable use-case of a client using such information. @maybeec and @SchettlerKoehler can we have a discussion about REST design to clarify all such things (see also my new review comments)? It seems rather heavy to discuss and solve such things in PR reviews...